stl/inc/chrono

```diff
@@ -2209,6 +2221,2949 @@ namespace chrono {
+            // Where "target_ls" is the next leap second at or after _Time, _It either points to:
+            //    (1) The 2nd leap second after _Time if _Time_floor is in the range [target_ls - _Elapsed() - 1,
+            //    target_ls), or
+            //    (2) The leap second just after _Time otherwise.
+            // Note that we can always use prev(_It) to determine whether _Time is *during* a leap second insertion,
+            // since that falls under case (2) above. However, when we fall under case (1), we need to execute an
+            // additional decrement to get the correct elapsed offset. For example, if leap seconds are inserted at
+            // seconds {100, 200, 300, 400}, we have:
+            //
+            //  UTC   sys   *_It
+            //   99    99    100
+            //  100     X    200
+            //  101   100    200
+            //  102   101    200
+            //  ...
+            //  199   198    200
+            //  200   199    300^
+            //  201     X    300
+            //  202   200    300
+            //  ...
+            //  299   297    300
+            //  300   298    400^
+            //  301   299    400^
+            //  302     X    400
+            //  303   300    400
+            //
+            //  ^_It points to 2nd leap second
+
+            _It = _STD upper_bound(_Ls_vector.begin(), _Ls_vector.end(), sys_seconds{_Time_floor.time_since_epoch()});
+        } else {
+            seconds _Prev_elapsed{0};
+            for (_It = _Ls_vector.begin(); _It != _Ls_vector.end(); ++_It) {
+                // UTC time when leap second insertion begins. In all cases, _It->date() + _It->_Elapsed() is the *end*
+                // of the insertion. For a negative leap that's also the beginning, but for a positive one, insertion
+                // begins 1 second earlier.
+                const utc_seconds _This_ls_begin{
+                    _It->date().time_since_epoch() + (_It->_Positive() ? _Prev_elapsed : _It->_Elapsed())};
+                if (_This_ls_begin > _Time_floor) {
+                    break;
+                }
+                _Prev_elapsed = _It->_Elapsed();
+            }
+        }
+
+        if (_It == _Ls_vector.begin()) {
+            return {false, seconds{0}};
+        } else {
+            // Convert to the last leap second before or equal to _Time.
+            const auto& _Last_leap = *--_It;
+            const utc_seconds _Utc_leap_second{_Last_leap.date().time_since_epoch() + _It->_Elapsed() - seconds{1}};
+#ifdef __cpp_lib_concepts
+            const auto _Leap_cmp = _Utc_leap_second <=> _Time_floor;
+#else // ^^^ __cpp_lib_concepts / TRANSITION, GH-395 workaround vvv
+            const auto _Leap_cmp = _Utc_leap_second > _Time_floor  ? strong_ordering::greater
+                                 : _Utc_leap_second == _Time_floor ? strong_ordering::equal
+                                                                   : strong_ordering::less;

```

Comment:
Nitpick, I find it a bit strange that we split around the `>` operation (And why are we using `>`)

Could we do the equals check first and then do a `<`
```suggestion
            const auto _Leap_cmp = _Utc_leap_second == _Time_floor ? strong_ordering::equal
                                 : _Utc_leap_second < _Time_floor  ? strong_ordering::less
                                                                   : strong_ordering::greater;
```

Reply:
It is slightly strange, but as this is the C++20-without-concepts workaround, which will hopefully be short-lived, I think that stylistically rearranging this code wouldn't be worth the (minor) time and risk.